src/App.js

```diff
@@ -1,20 +1,441 @@
+       * id: newId modifies the id property of the shallow copy and assignes the value of `newId` to the `id` property of the copied object
+       */
+
+      this.setState(
+        (prevState) => ({
+          // Wrap the arrow function in () to represent object literal and not function block!
+          tasks: prevState.tasks.filter((task) => task.id !== id),
+          tasksInProgress: [...prevState.tasksInProgress, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // Transfer tasks from Tasks In Progress to Open Tasks
+  moveTaskInProgressBacktoOpen = (id) => {
+    const taskToMove = this.state.tasksInProgress.find(
+      (task) => task.id === id
+    ); // finds task matching the id passed into the argument using find() method
+
+    if (taskToMove) {
+      const newId = this.state.tasks.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+
+      this.setState(
+        (prevState) => ({
+          tasksInProgress: prevState.tasksInProgress.filter(
+            (task) => task.id !== id
+          ),
+          tasks: [...prevState.tasks, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // Transfer tasks from In Review back to In Progress
+  moveTaskInReviewBackToInProgress = (id) => {
+    const taskToMove = this.state.tasksInReview.find((task) => task.id === id);
+
+    if (taskToMove) {
+      const newId = this.state.tasksInProgress.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+
+      this.setState(
+        (prevState) => ({
+          tasksInReview: prevState.tasksInReview.filter(
+            (task) => task.id !== id
+          ),
+          tasksInProgress: [...prevState.tasksInProgress, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // Transfer tasks from In Progress to In Review
+  moveTaskInProgress = (id) => {
+    const taskToMove = this.state.tasksInProgress.find(
+      (task) => task.id === id
+    );
+    if (taskToMove) {
+      const newId = this.state.tasksInReview.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+      this.setState(
+        (prevState) => ({
+          tasksInProgress: prevState.tasksInProgress.filter(
+            (task) => task.id !== id
+          ),
+          tasksInReview: [...prevState.tasksInReview, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // OPEN TASKS SECTION
+  updateTask = (id, updatedTask) => {
+    this.setState((prevState) => {
+      const updatedTasks = prevState.tasks.map((task) => {
+        if (task.id === id) {
+          return { ...task, ...updatedTask };
+        }
+        return task;
+      });
+      return { tasks: updatedTasks };
+    }, this.updateLocalStorage);
+  };
+
+  addTaskToDo = (task) => {
+    const newId = this.state.tasks.length;
```

Comment:
I think this might possibly be buggy?
Let's say I got these tasks: `[1,2,3,4,5,6]`

I then go ahead and delete task `2`: `[1,3,4,5,6]`

Wouldn't I create another task `6` on the next create? `[1,3,4,5,6,6]`

Please correct me if I am wrong.

Reply:
I think should rather be `[1,3,4,5,6,5]` then

Reply:
A recommendation from my side would be to use a UUID. 
https://www.npmjs.com/package/uuid